validphys2/src/validphys/commondataparser.py

```diff
@@ -89,3 +92,87 @@ def parse_systypes(systypefile):
     systypetable.set_index("sys_index", inplace=True)
 
     return systypetable
+
+def write_commondata(commondata_list, filter_path):
+    """
+    GENERAL DESCRIPTION:
+
+    writes dataset data in filter folder using the commondata file format
+
+    Parameters
+    ----------
+
+    commondata_list : list
+                     commondata object (cuts should be already applied)
+
+    path : str
+         path to filter folder
+
+    
+    """
+
+    for commondata_instance in commondata_list:
+        # path
+        path = str(filter_path) + f'/{commondata_instance.setname}'
+        path_data = str(path) + f"/DATA_{commondata_instance.setname}.dat"
+        commondata_tab = commondata_instance.commondata_table.reset_index(drop = True) # do not use maskcut index
+        commondata_tab.index += 1 # index starting from 1
+        
+        header = f"{commondata_instance.setname} {commondata_instance.nsys} {commondata_instance.ndata}\n"
+        
+        #==== write DATA =====#
+        with open(path_data, "w+") as f:
```

Comment:
Would be good to have these this as a separate function, e.g. `def write_commondata_data(commondata, buffer)` that could be used as

```
from io import StringIO

sio = StrinIO()
write_commondata_data(commondata, sio)
print(sio.get_value()) # Contents of a commondata file
```

also, you don't want to have the `"w+"` mode, but the `"w"` one
    
```

Reply:
Ok, I am not sure I completely understand what you mean. Would you rather prefer to have something like:

```
def write_commondata_pd(commondata,buffer):
       commondata_table = commondata.commondata_table
       commondata_table.to_csv(buffer, sep = "\t",header=None)
```

```
def write_commondata_to_file(commondata,path):
    from io import StringIO
    
    sio = StringIO()
    write_commondata_pd(commondata,sio)
    
    header = f"{commondata.setname} {commondata.nsys} {commondata.ndata}\n"
    
    with open(path,"w") as file:
        file.write(header)
        file.write(sio.getvalue())
```

Reply:
I’d like something that  writes the whole commondata file to an arbitrary buffer that could be an open file, a stringio object or anything else.